app/src/main/java/org/stepik/android/view/course_complete/ui/dialog/CourseCompleteBottomSheetDialogFragment.kt

```diff
@@ -0,0 +1,620 @@
+import ru.nobird.android.view.redux.ui.extension.reduxViewModel
+import javax.inject.Inject
+
+class CourseCompleteBottomSheetDialogFragment : BottomSheetDialogFragment(),
+    ReduxView<CourseCompleteFeature.State, CourseCompleteFeature.Action.ViewAction> {
+    companion object {
+        const val TAG = "CourseCompleteBottomSheetDialogFragment"
+
+        const val ARG_COURSE = "course"
+
+        fun newInstance(course: Course): DialogFragment =
+            CourseCompleteBottomSheetDialogFragment().apply {
+                this.course = course
+            }
+    }
+
+    @Inject
+    internal lateinit var screenManager: ScreenManager
+
+    @Inject
+    internal lateinit var viewModelFactory: ViewModelProvider.Factory
+
+    @Inject
+    internal lateinit var shareHelper: ShareHelper
+
+    private var course: Course by argument()
+    private val viewModel: CourseCompleteViewModel by reduxViewModel(this) { viewModelFactory }
+
+    private lateinit var viewStateDelegate: ViewStateDelegate<CourseCompleteFeature.State>
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        injectComponent()
+        setStyle(DialogFragment.STYLE_NO_TITLE, R.style.TopCornersRoundedBottomSheetDialog)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? =
+        inflater.inflate(R.layout.bottom_sheet_dialog_course_complete, container, false)
+
+    private fun injectComponent() {
+        App.component()
+            .courseCompleteComponentBuilder()
+            .build()
+            .inject(this)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        courseCompleteFeedback.background = AppCompatResources
+            .getDrawable(requireContext(), R.drawable.bg_shape_rounded)
+            ?.mutate()
+            ?.let { DrawableCompat.wrap(it) }
+            ?.also {
+                DrawableCompat.setTint(it, ContextCompat.getColor(requireContext(), R.color.color_overlay_violet_alpha_12))
+                DrawableCompat.setTintMode(it, PorterDuff.Mode.SRC_IN)
+            }
+        viewStateDelegate = ViewStateDelegate()
+        viewStateDelegate.addState<CourseCompleteFeature.State.Idle>()
+        viewStateDelegate.addState<CourseCompleteFeature.State.Loading>(courseCompleteProgressbar)
+        viewStateDelegate.addState<CourseCompleteFeature.State.Content>(
+            courseCompleteHeader,
+            courseCompleteTitle,
+            courseCompleteFeedback,
+            courseCompleteSubtitle,
+            viewCertificateAction,
+            shareResultAction,
+            courseCompleteDivider,
+            primaryAction,
+            secondaryAction
+        )
+        viewStateDelegate.addState<CourseCompleteFeature.State.NetworkError>()
```

Comment:
Is it ok to ignore error state?

Reply:
Will dismiss dialog and show snackbar if this state is set